spec/lib/jobs/switch-node-relations-spec.js

```diff
@@ -431,11 +455,35 @@ describe(require('path').basename(__filename), function () {
                     }
                 ]
             };
+            var switchNode = {
+                id: nodeId,
+                relations: [
+                    {
+                        'relationType': 'connects',
+                        'targets': ['existingnodeId']
+                    }
+                ]
+            };
+
+            var switchRelation = {
+                relations: [
+                    {
+                        'relationType': 'connects',
+                        'targets': ['existingnodeId', 'nodeid']
+                    }
+                ]
+            };
+
+            needByIdentifier.resolves(switchNode);
             findByIdentifier.resolves(node);
-            job.run()
+            return job.run()
             .then(function () {
                 expect(mockWaterline.nodes.updateByIdentifier)
-                .to.have.been.calledWith('nodeid', nodeRelation);
+                .to.have.been.calledWith('nodeid', nodeRelation)
+                .to.have.been.calledWith(nodeId, switchRelation);
+            })
+            .catch(function (err) {
```

Comment:
@yyscamper Comments addressed. Regarding your comments about if the code works; 
check : https://github.com/RackHD/on-tasks/pull/346/files#diff-61bb15c9eb0a1698c8d46428ec4df11dR161 this is where i make sure I eliminate duplicates.
Thanks